include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -220,6 +213,21 @@ class PreconditionerBoomerAMG : public PreconditionerBase<Number>
                             });
   }
 
+  void
+  update() override
+  {
+    // clear content of matrix since the next calculate_system_matrix calls
+    // add their result; since we might run this on a sub-communicator, we
+    // skip the processes that do not participate in the matrix and have size
+    // zero
+    if(system_matrix.m() > 0)
+      system_matrix = 0.0;
```

Comment:
why not `*= 0` as above?

Reply:
I would suggest to look at this somewhere else, or did I change the code here in this PR?

Reply:
no, code was like this before - I was just wondering.
We could open up an issue and see what the others think?